2check/app/components/character/character.component.ts

```diff
@@ -0,0 +1,320 @@
+    let result: number = firstValue;
+    if (better == firstValue) {
+      result = secondValue;
+    }
+    return result;
+  }
+
+  damageCheck(health: number, dmg: number, isCriticalRoll: boolean = false) {
+    if (isCriticalRoll || Math.round(health / 2) < dmg) {
+      this.panicCheck();
+    }
+    health -= dmg;
+    if (health <= 0) {
+      this.dead = true;
+    }
+    return health;
+  }
+
+  armorCheck(armor: number, dmg: number, isCriticalRoll: boolean = false) {
+    if (isCriticalRoll) {
+      armor = armor / 2;
+    }
+    dmg -= armor;
+    dmg = (dmg < 0) ? 0 : dmg;
+    return dmg;
+  }
+
+  saveCheck(saveType: string, saveValue: number, advantage: boolean = false, disadvantage: boolean = false) {
+    let passed: boolean = false;
+    let critical: boolean = false;
+    let firstTry: number;
+    let secondTry: number;
+    let result: number;
+
+    if (advantage > disadvantage) {
+      firstTry = +prompt(`${this.character.name} preforming ${saveType} check with advantage! Roll d100 under/equal to the ${saveValue} to pass save check!`);
+      secondTry = +prompt(`${this.character.name} preforming second trow!`);
+      result = this.betterValue(saveValue, firstTry, secondTry);
+    } else if (advantage < disadvantage) {
+      firstTry = +prompt(`${this.character.name} preforming ${saveType} check with Disadvantage! Roll d100 under\equal to ${saveValue} to pass save check!`);
+      secondTry = +prompt(`${this.character.name} preforming second trow!`);
+      result = this.worstValue(saveValue, firstTry, secondTry)
+    } else {
+      result = +prompt(`${this.character.name} preforming ${saveType} check! Roll d100 under/equal to the ${saveValue} to pass save check!`);
+    }
+
+    critical = this.isCriticalRoll(result);
+
+    if (result >= saveValue) {
+      if (critical) {
+        alert(`${saveType} check - CRITICAL FAIL with ${result} trow!`);
+        this.panicCheck();
+      } else {
+        this.character.stress += 1;
+        alert(`${saveType} check - FAIL with ${result} trow! 1 stress point gained`);
+      }
+    } else {
+      passed = true;
+      if (critical) {
+        alert(`WOW! ${saveType} check - CRITICAL SUCCESS with ${result} trow!`);
+      } else {
+        alert(`${saveType} check - SUCCESS with ${result} trow!`);
+      }
+    }
+    return passed;
+  }
+
+  sanitySave() {
+    this.saveCheck('SANITY', this.character.saves.Sanity, this.character.advantages.Sanity, this.character.disadvantages.Sanity);
+  }
+
+  fearSave() {
+    return this.saveCheck('FEAR', this.character.saves.Fear, this.character.advantages.Fear, this.character.disadvantages.Fear);
+  }
+
+  bodySave() {
+    return this.saveCheck('BODY', this.character.saves.Body, this.character.advantages.Body, this.character.disadvantages.Body);
+  }
+
+  armorSave() {
+    return this.saveCheck('ARMOR', this.character.saves.Armor, this.character.advantages.Armor, this.character.disadvantages.Armor);
+  }
+
```

Comment:
generally, pcharacter.component.ts is place where you place logic that belongs to character visualization, but here i see a lot of game mechanics stuff.
Business logic should be places in services, ui logic in components